src/SFA.DAS.Admin.Aan.Web/Validators/ManageEvent/EventDateTimeViewModelValidator.cs

```diff
@@ -0,0 +1,76 @@
+{
+    public const string EventDateInPast = "Event date must not be in the past";
+    public const string EventDateEmpty = "You must select an event date";
+    public const string EventStartHourEmpty = "You must select a start time hour";
+    public const string EventStartMinutesEmpty = "You must select a start time minutes";
+    public const string EventStartHourAndMinutesEmpty = "You must select a start time hour and minutes";
+    public const string EventEndHourEmpty = "You must select an end time hours";
+    public const string EventEndMinutesEmpty = "You must select an end time minutes";
+    public const string EventEndHourAndMinutesEmpty = "You must select an end time hour and minutes";
+    public const string EventEndTimeBeforeStartTime = "Event end time is before event start time";
+
+    public EventDateTimeViewModelValidator()
+    {
+        RuleFor(x => x.DateOfEvent)
+            .NotEmpty()
+            .WithMessage(EventDateEmpty)
+            .GreaterThanOrEqualTo(DateTime.Today)
+            .WithMessage(EventDateInPast);
+
+        RuleFor(dateTime => dateTime.StartHour)
+             .Cascade(CascadeMode.Stop)
+             .Must(StartHoursAndMinutesPresent)
+             .WithMessage(EventStartHourAndMinutesEmpty)
+             .NotEmpty()
+             .WithMessage(EventStartHourEmpty)
+             .Must(StartTimePresent)
+             .WithMessage(EventStartMinutesEmpty);
+
+        RuleFor(dateTime => dateTime.EndHour)
+            .Cascade(CascadeMode.Stop)
+            .Must(EndHoursAndMinutesPresent)
+            .WithMessage(EventEndHourAndMinutesEmpty)
+            .NotEmpty()
+            .WithMessage(EventEndHourEmpty)
+            .Must(EndTimePresent)
+            .WithMessage(EventEndMinutesEmpty)
+            .Must(StartTimeBeforeEndTime)
+            .WithMessage(EventEndTimeBeforeStartTime);
+    }
+
+    private bool StartHoursAndMinutesPresent(EventDateTimeViewModel model, int? startHour)
+    {
+        return !(model.StartMinutes == null && startHour == null);
+    }
+
+    private bool StartTimePresent(EventDateTimeViewModel model, int? startHour)
+    {
+        return model.StartMinutes != null;
+    }
+
+    private bool EndHoursAndMinutesPresent(EventDateTimeViewModel model, int? endHour)
+    {
+        return !(model.EndMinutes == null && endHour == null);
+    }
+
+    private bool EndTimePresent(EventDateTimeViewModel model, int? endHour)
+    {
+        return model.EndMinutes != null;
+    }
+
+    private bool StartTimeBeforeEndTime(EventDateTimeViewModel model, int? startHour)
+    {
+        if (model.StartHour == null || model.StartMinutes == null || model.EndHour == null ||
+            model.EndMinutes == null) return true;
+
+        var startTime = (model.StartHour * 60) + model.StartMinutes;
+        var endTime = (model.EndHour * 60) + model.EndMinutes;
+        return startTime <= endTime;
```

Comment:
There are 8 conditions not covered in tests;
 
    private bool StartTimeBeforeEndTime(EventDateTimeViewModel model, int? startHour)
    {
        if (model.StartHour == null || model.StartMinutes == null || model.EndHour == null ||
            model.EndMinutes == null) return true;
        var startTime = (model.StartHour * 60) + model.StartMinutes;
        var endTime = (model.EndHour * 60) + model.EndMinutes;
        return startTime <= endTime;
    }